### PR TITLE
Draft for NXmpes_arpes application definition

### PR DESCRIPTION
**Pull request by rettigl**
This is a draft for an NXmpes_arpes subdefinition. It is not adjusted yet to conform with changes introduced into NXmpes within the refactor process, and some of the concepts still need to be ported to NXmpes.

It assumes angular-dependent measurements with an angular resolving detector, and allows for flexible acquisition modes. The data should contain two angular and one energy coordinate.

The pyarpes loader here https://github.com/FAIRmat-Experimental/arpes/pull/1 supports loading files generated with this Appdef, and converting into momentum space. A respective tutorial is still pending.

### CODE REVIEW

File: contributed_definitions/nyaml/NXmpes_arpes.yaml [hunk 1]
```diff
@@ -0,0 +1,152 @@
+#Pincelli, Rettig, Arora at fhi-berlin.mpg.de, Dobener at hu-berlin.de, 06/2022
+#Draft version of a NeXus application definition for angle resolved photoemission.
+#This appdef aims to describe data produced from hemispherical anlysers,
```
Comment (lukaspie) [inline contributed_definitions/nyaml/NXmpes_arpes.yaml:3]:
```suggestion
#This appdef aims to describe data produced from hemispherical analysers,
```

File: contributed_definitions/nyaml/NXmpes_arpes.yaml [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+#with at least an angular coordinate and one an energy coordinate.
```
Comment (lukaspie) [inline contributed_definitions/nyaml/NXmpes_arpes.yaml:4]:
```suggestion
#with at least an angular coordinate and one energy coordinate.
```

File: contributed_definitions/nyaml/NXmpes_arpes.yaml [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+            exists: recommended
+            doc: "Raw data before calibration."
+    (NXprocess):
+      exists: optional
+      doc: "Document an event of data processing, reconstruction, or analysis for this data.
+        Describe the appropriate axis calibrations for your experiment using
+        one or more of the following NXcalibrations"
+      energy_calibration(NXcalibration):
+        exists: optional
+        applied(NX_BOOLEAN):
```
Comment (lukaspie) [inline contributed_definitions/nyaml/NXmpes_arpes.yaml:84]: `applied` can be removed in accordance with NXmpes.

File: contributed_definitions/nyaml/NXmpes_arpes.yaml [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+          doc: "Has an energy calibration been applied?"
+        calibrated_axis(NX_FLOAT):
+          exists: recommended
+          doc: "This is the calibrated energy axis to be used for data plotting."
+      angular1_calibration(NXcalibration):
+        exists: optional
+        applied(NX_BOOLEAN):
```
Comment (lukaspie) [inline contributed_definitions/nyaml/NXmpes_arpes.yaml:91]: see above

File: contributed_definitions/nyaml/NXmpes_arpes.yaml [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+          doc: "Has an angular calibration been applied to dispersive axis 1?"
+        calibrated_axis(NX_FLOAT):
+          exists: recommended
+          doc: "This is the calibrated angular axis to be used for data plotting."
+      angular2_calibration(NXcalibration):
+        exists: optional
+        applied(NX_BOOLEAN):
```
Comment (lukaspie) [inline contributed_definitions/nyaml/NXmpes_arpes.yaml:98]: see above

**Review by lukaspie — CHANGES_REQUESTED** [review]
Currently just typo fixes and things to remove, will review again when NXmpes is finalized.

**Comment by lukaspie** [comment]
For your convenience, I have merged `mpes_refactor` and recreated the NXDL files for the changed nyamls using [nyaml v0.0.4](https://github.com/FAIRmat-NFDI/nyaml).

**Review by lukaspie — APPROVED** [review]
I cannot really comment much on the transformations, but the rest looks good to me.

**Comment by rettigl** [comment]
@lukaspie @domna Is there any blocker for merging this?

**Comment by lukaspie** [comment]
> @lukaspie @domna Is there any blocker for merging this?

The only remaining issue I see is that
```
geometries(NXcoordinate_system_set):
  arpes_geometry(NXcoordinate_system):
    depends_on(NX_CHAR):
      doc: |
        Link to transformations defining an APRES base coordinate system,
        which is defined such that the positive z-axis points towards the analyzer entry,
        and the x-axis lies within the beam/analyzer plane.
    (NXtransformations):
      doc: |
        Set of transformations, describing the orientation of the ARPES coordinate system
        with respect to the beam coordinate system (.).
```
is currently not supported since the base class `NXcoordinate_system` does not contain `NXtransformations` yet. They are to be added by https://github.com/FAIRmat-NFDI/nexus_definitions/pull/144. My plan was actually to check that PR today and finalize it. But, since we already agreed that #144 will be accepted, we can also merge this PR here first.

**Comment by rettigl** [comment]
> is currently not supported since the base class `NXcoordinate_system` does not contain `NXtransformations` yet. They are to be added by #144. My plan was actually to check that PR today and finalize it. But, since we already agreed that #144 will be accepted, we can also merge this PR here first.

Even without the additions in the base class, this will be a valid definition, just with the respective elements in the appdef...

I think this is ready for merge, please do so if you agree.

**Comment by rettigl** [comment]
@lukaspie @domna I would suggest to merge this now, close #231 (integrated here), and also finalize #144

**Comment by lukaspie** [comment]
Let's merge this for now. I am workshopping an example for using NXcoordinate_system and will present it in one of the coming TF meetings. So, depending on the discussion, we may have to slightly change this definition as well. But we can always do this later.

**Pull request merged**